react-toggle-switch-component/src/toggle.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+class Toggle extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      toggle: false,
+      color: '',
+      onOrOff: ''
+    };
+    this.handleToggle = this.handleToggle.bind(this);
+  }
+
+  handleToggle() {
+    this.setState({ toggle: !this.state.toggle });
+  }
+
+  render() {
+    const $body = document.querySelector('body');
+    let { toggle, color, onOrOff } = this.state;
+    if (toggle === false) {
+      onOrOff = 'OFF';
+      $body.className = '';
+    } else {
+      onOrOff = 'ON';
+      color = 'white';
+      $body.className = 'dark';
+      toggle = false;
+    }
```

Comment:
If you never need to `setState` a value that's in a component's state, that probably means it doesn't need to be in state in the first place.